operator/transformer/recombine/recombine.go

```diff
@@ -220,17 +224,26 @@ func (r *RecombineOperator) Process(ctx context.Context, e *entry.Entry) error {
 		s = DefaultSourceIdentifier
 	}
 
-	// This is the first entry in the next batch
-	if matches && r.matchIndicatesFirst() {
-		// Flush the existing batch
-		err := r.flushSource(s)
-		if err != nil {
-			return err
+	// For first entry verification
+	if r.matchIndicatesFirst() {
+		// If entry matches, set firstLineMatched for source to true
+		if !r.firstLineMatched[s] && matches {
+			r.firstLineMatched[s] = true
 		}
 
-		// Add the current log to the new batch
-		r.addToBatch(ctx, e, s)
-		return nil
+		// This is the first entry in the next batch
+		// or any entry if there wasn't any matching entry before
+		if matches || !r.firstLineMatched[s] {
+			// Flush the existing batch
+			err := r.flushSource(s)
+			if err != nil {
+				return err
+			}
+
+			// Add the current log to the new batch
+			r.addToBatch(ctx, e, s)
+			return nil
+		}
```

Comment:
Do we really need to keep track of `firstLineMatched`? 

Can we just make sure never to start a batch with a partial (when matching on first line)?

```go
switch {
case matches && r.matchIndicatesFirst():
	... 
	return nil
case matches && r.matchIndicatesLast():
	... 
	return nil
case !matches && r.matchIndicatesFirst():
	if len(batchMap[s]) == 0 {
		// When matching on first entry, never batch partial first. Just emit immediately
		r.Write(context.Background(), e)
		return nil
	}
}
r.addToBatch(ctx, e, s)
```

Reply:
Yees, it's much better
Implemented :)